workers/utils.py

```diff
@@ -324,11 +324,26 @@ def setup_environ() -> None:
 FUZZER_CMD_PORT = 7055
 
 class _PausedFuzzers:
+    def _perform_request(self, action):
+        """Perform an action (pause or resume) without risking an exception"""
+        last_exception = None
+        for n in range(3):
+            try:
+                requests.get(f'http://127.0.0.1:{FUZZER_CMD_PORT}/{action}')
+                return
+            except requests.exceptions.RequestException as ex:
+                last_exception = ex
+                time.sleep(1.5 ** n)
+        print(f'Failed to {action} fuzzers: {last_exception}', file=sys.stderr)
+
     def __init__(self):
-        requests.get(f'http://127.0.0.1:{FUZZER_CMD_PORT}/pause')
+        self._perform_request('pause')
+
+    def __enter__(self):
+        pass
```

Comment:
`pause` should happen in `__enter__`.

Reply:
Hmm I used the same logic as the one given as example [here](https://book.pythontips.com/en/latest/context_managers.html#implementing-a-context-manager-as-a-class): perform the action in `__init__` and just return what the user may want to use from `__enter__`. Did I misunderstand python's conventions because I only looked at this example?

Reply:
The issue is that currently if this is not used as a context manager than processing will be paused in __init__ but will never be resumed.